lib/puppet/type/posix_acl.rb

```diff
@@ -161,8 +162,10 @@ def unset_insync(cur_perm)
       (sp - cp).sort == sp
     end
 
+    # Make sure we are not misinterpreting recursive permission notation (e.g. rwX) when
+    # comparing current to new perms.
     def set_insync(cur_perm) # rubocop:disable Style/AccessorMethodName
-      should = @should.uniq.sort
+      should = @should.uniq.map(&:downcase).sort
```

Comment:
perhaps better to first downcase and then uniq

Reply:
Excellent point  =D  Changed!